src/main/java/com/shopify/mappers/LegacyToFulfillmentOrderMapping.java

```
@@ -0,0 +1,170 @@
+			// referenced inside the payload's line_items_by_fulfillment_order
+			// section
+			for (final ShopifyFulfillmentOrder fulfillmentOrder : fulfillmentOrders) {
+				// if a shopify fulfillment is cancelled a new fulfillment order
+				// is added the old one gets it's supported actions emptied, so
+				// we need to make sure the current fulfillmentOrder supportes
+				// fulfillment creation under it
+				if (fulfillmentOrder.getSupportedActions().contains(SupportedActions.CREATE_FULFILLMENT.toString())) {
+					ShopifyLineItemsByFulfillmentOrder lineItemsByFulfillment = new ShopifyLineItemsByFulfillmentOrder();
+					lineItemsByFulfillment.setFulfillmentOrderId(fulfillmentOrder.getId());
+					for (final ShopifyFulfillmentOrderLineItem fulfillmentOrderLineItem : fulfillmentOrder
+							.getLineItems()) {
+						for (final ShopifyLineItem fulfillmentLineItem : fulfillment.getLineItems()) {
+							if (fulfillmentOrderLineItem.getLineItemId().equals(fulfillmentLineItem.getId())) {
+								ShopifyFulfillmentOrderPayloadLineItem payloadLineItem = new ShopifyFulfillmentOrderPayloadLineItem(
+										fulfillmentOrderLineItem.getId(), fulfillmentLineItem.getQuantity());
+								lineItemsByFulfillment.getFulfillmentOrderLineItems().add(payloadLineItem);
+							}
+						}
+					}
+
+					if (lineItemsByFulfillment.getFulfillmentOrderLineItems().size() > 0) {
+						lineItemsByFulfillmentOrder.add(lineItemsByFulfillment);
+					}
+				}
+			}
+
+			payload.setTrackingInfo(trackingInfo);
+			payload.setNotifyCustomer(fulfillment.isNotifyCustomer());
+			payload.setLineItemsByFulfillmentOrder(lineItemsByFulfillmentOrder);
+
+			ShopifyFulfillmentPayloadRoot root = new ShopifyFulfillmentPayloadRoot();
+			root.setFulfillment(payload);
+			return root;
+		} catch (final Exception e) {
+			LOGGER.error("There was an error parsing fulfillmentOrder into a ShopifyFulfillment create payload");
+			return null;
```

Same thing here. let's bubble the exception up and also add cause to the log